roles/scm/git/tasks/pull.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: Prepare SCM credentials
+  include_tasks: prep.yml
+
+- name: Create temporary directory for the repository
+  tempfile:
+    state: directory
+    prefix: "repo-"
+  register: result_scm_dir
+  changed_when: false
+  when:
+    - not repository.path is defined
+
+- name: Set directory for the repository
+  set_fact:
+    scm_dir: "{{ repository.path if repository.path is defined else result_scm_dir.path }}"
+
+- name: "Clone repository to temporary location"
+  git:
+    clone: true
+    repo: "{{ scm_repository }}"
+    dest: "{{ scm_dir }}"
+    force: "{{ repository.force | bool | default(omit) }}"
```

Comment:
the `bool` flag here won't work as it will complain about `undefined variable` if `force` isn't set. My suggestion would be to just drop the flag and rely on the user to pass the right value

Reply:
same below where the `boot | default(omit)` combination is used